### PR TITLE
Remove deprecated fields from compiler

### PR DESCRIPTION
**Pull request by ashawley**
These seemed like low risk things to trim. I am not an expert.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/util/package.scala [hunk 1]
```diff
@@ -116,17 +116,6 @@ package object util {
 
   lazy val trace = new SimpleTracer(System.out)
 
-  // These four deprecated since 2.10.0 are still used in (at least)
-  // the sbt 0.12.4 compiler interface.
-  @deprecated("Moved to scala.reflect.internal.util.Position", "2.10.0")
-  type Position = scala.reflect.internal.util.Position
```
Comment (eed3si9n) [inline src/compiler/scala/tools/nsc/util/package.scala]: Removing `Position` may prevent the `compile` task in sbt 0.13 - https://github.com/sbt/sbt-zero-thirteen/blob/v0.13.17/compile/interface/src/main/scala/xsbt/DelegatingReporter.scala#L18

Comment (ashawley) [reply src/compiler/scala/tools/nsc/util/package.scala]: Ok, I'll put back and update the version in the comment to sbt 0.13.

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/util/package.scala]: Or possibly 2.13 is a good moment to drop support for 0.13.

Comment (SethTisue) [reply src/compiler/scala/tools/nsc/util/package.scala]: > Or possibly 2.13 is a good moment to drop support for 0.13

I wish it were, but I don't think so

Comment (ashawley) [reply src/compiler/scala/tools/nsc/util/package.scala]: Agree. Could be something to aim for in 2.14, though.

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/util/package.scala]: Maybe 0.13 could drop support for 2.13 instead.

Comment (ashawley) [reply src/compiler/scala/tools/nsc/util/package.scala]: Or release 0.14 as an EOL release: It doesn't support any Scala version.  End of line.

Comment (ashawley) [reply src/compiler/scala/tools/nsc/util/package.scala]: Or release 0.14 as an EOL release: It doesn't support any Scala version.  End of line.

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/util/package.scala]: @ashawley should definitely write the next April 1 release notes.

Comment (dwijnand) [reply src/compiler/scala/tools/nsc/util/package.scala]: > I wish it were, but I don't think so

Given sbt 0.13 completed its EOL before Scala 2.13 was released, I think it's fair to change things in Scala 2.13 even if it breaks sbt 0.13.  That said, I don't think this change is significant enough to start that process.

Comment (eed3si9n) [reply src/compiler/scala/tools/nsc/util/package.scala]: I'm advocating for keeping sbt 0.13 support in part for the interest of Scala 2.13. As an sbt maintainer, I am happy to say "just use latest sbt 1 patch." But assuming some percentage (source needed) of the OSS and enterprise world builds are on sbt 0.13, _not_ having sbt 0.13 would be an additional hurdle to Scala 2.13 adoption.

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/util/package.scala]: OTOH, having a high bar keeps them from complaining about the other hurdles to adoption.

Comment (sjrd) [reply src/compiler/scala/tools/nsc/util/package.scala]: Just so you know: Scala.js uses sbt 0.13.17 ... ScalaTest 3.0.x as well.

**Comment by lrytz** [comment]
I'd move this out of RC2, there's no functional benefit, but some risk for breaking something.

**Comment by diesalbla** [comment]
@ashawley Considering the recent #8042 , are you still pushing for this one?

**Comment by ashawley** [comment]
@diesalbla Yes, I'm still pushing it, but this has been considered too risky for 2.13.0-RC2 and so is scheduled for 2.13.1.

I moved  the less risky parts from here to  #8042.

**Comment by som-snytt** [comment]
They don't have an emoji reaction for less risky. 😌 ?

File: src/compiler/scala/tools/nsc/util/package.scala (hunk 1 above)
Comment (ashawley) [reply src/compiler/scala/tools/nsc/util/package.scala]: Updating the comment about sbt was split out in to #8042.

**Comment by szeiger** [comment]
@lrytz Should we merge this or wait until 2.14?

**Comment by szeiger** [comment]
We discussed this in today's team meeting. There's no real benefit from removing a few deprecated forwarders but a potential to break compiler plugins and tooling integration. Rescheduling for 2.14 for now, but maybe we should forego this entirely.

**Comment by ashawley** [comment]
I understand the concern about breakage, so fine to move to 2.14.  

The motivation is that these are removed from Dotty, so it could help with the transition, FWIW.

**Comment by SethTisue** [comment]
It's better to just leave this alone, I think (especially after https://www.scala-lang.org/2019/12/18/road-to-scala-3.html)

**Pull request closed without merging**